### PR TITLE
RSDK-8699: Cannot go to gantry position 0.0, only 0.2+.

### PR DESCRIPTION
**Pull request by martha-johnston**
- remove error margin to allow gantry to move to 0
- return more logs and errors for frontend

### CODE REVIEW

**Review by penguinland — APPROVED** [review]
LGTM! I like how straightforward the code changes are: it does exactly what the PR title says.

**Review by oliviamiller — APPROVED** [review]

**Comment by martha-johnston** [comment]
> LGTM! I like how straightforward the code changes are: it does exactly what the PR title says.

lol thanks the last minute bucket review

**Comment by martha-johnston** [comment]
@michaellee1019 if you're interested

File: components/gantry/singleaxis/singleaxis_test.go [hunk 1]
```diff
@@ -868,10 +868,10 @@ func TestGoToInputs(t *testing.T) {
 
 	inputs = []referenceframe.Input{{Value: 1.0}}
 	err = fakegantry.GoToInputs(ctx, inputs)
-	test.That(t, err, test.ShouldBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "cannot move past limit switch")
 
 	err = fakegantry.GoToInputs(ctx, inputs, inputs)
-	test.That(t, err, test.ShouldBeNil)
+	test.That(t, err.Error(), test.ShouldContainSubstring, "cannot move past limit switch")
```
Comment (randhid) [inline components/gantry/singleaxis/singleaxis_test.go:874]: Why did the behaviour of these tests change?

**Review by randhid — CHANGES_REQUESTED** [review]
Please add tests to ensure we can move to 0 and the limit margin, this way we can verify that these cases are maintained  in the future.

**Comment by martha-johnston** [comment]
> Please add tests to ensure we can move to 0 and the limit margin, this way we can verify that these cases are maintained in the future.

Added a test to 0, but limit margin is removed so we shouldn't need a test for that!

**Comment by martha-johnston** [comment]
> We should test that we can go to the end of the limit - PositionB, but error if we go further.

added a test for going exactly to position b, but tests for going further already exist so I didn't add any more

**Review by randhid — APPROVED** [review]

**Pull request merged**